climada_petals/engine/warn.py

```diff
@@ -0,0 +1,191 @@
+            if self.filter_data.expand:  # select points where level is >= level under observation -> expands regions
+                pts_curr_lvl = np.bitwise_or(warn_reg, d >= curr_lvl)
+            else:  # select points where level is == level under observation -> not expanding regions
+                pts_curr_lvl = d == curr_lvl
+            reg_curr_lvl = np.where(pts_curr_lvl, curr_lvl, 0)  # set everything but pts of current level to 0
+
+            for i in range(len(self.filter_data.operations)):
+                if self.filter_data.operations[i] == 'DILATION':
+                    reg_curr_lvl = skimage.morphology.dilation(reg_curr_lvl,
+                                                               skimage.morphology.disk(self.filter_data.sizes[i]))
+                elif self.filter_data.operations[i] == 'EROSION':
+                    reg_curr_lvl = skimage.morphology.erosion(reg_curr_lvl,
+                                                              skimage.morphology.disk(self.filter_data.sizes[i]))
+                elif self.filter_data.operations[i] == 'MEDIANFILTERING':
+                    filter_med = np.ones((self.filter_data.sizes[i], self.filter_data.sizes[i]))
+                    reg_curr_lvl = skimage.filters.median(reg_curr_lvl, filter_med)
+                else:
+                    LOGGER.warning("The operation is not defined. "
+                                   "Please select 'EROSION', 'DILATION', or 'MEDIANFILTERING'.")
+            return reg_curr_lvl
+
+        max_warn_level = np.max(d_thrs)
+        if max_warn_level == 0:
+            return np.zeros(d_thrs.shape)
+
+        warn_regions = 0
+        # iterate from highest level to lowest (0 not necessary, because rest is level 0)
+        for j in range(max_warn_level, 0, -1):
+            w_l = filtering(d_thrs, warn_regions, j)
+            # keep warn regions of higher levels by taking maximum
+            warn_regions = np.maximum(warn_regions, w_l)
+        self.warning = warn_regions
+
+    @staticmethod
+    def remove_small_regions(warning, size_thr):
+        """Remove regions smaller than defined threshold from warning.
+
+        Parameters
+        ----------
+        warning : np.array
+            Warning where regions should be removed from.
+        size_thr : int
+            Threshold defining too small regions (number of grid points).
+
+        Returns
+        ----------
+        warning : np.array
+            Warning without too small regions.
+        """
+
+        def increase_levels(warn, size):
+            # increase levels of too small regions to max level of this warning
+            labels = skimage.measure.label(warn)
+            for i in range(np.max(labels)):
+                cnt = np.count_nonzero(labels == i)
+                if cnt <= size:
+                    warn[labels == i] = np.max(warn, axis=(0, 1))
+            return warn
+
+        def set_new_lvl(warn, size):
```

Comment:
This method is unclear from its name. Make a proper function and add a docstring.